readthedocs/restapi/views/footer_views.py

```diff
@@ -83,39 +83,44 @@ def footer_html(request):
     else:
         print_url = None
 
-    show_promo = getattr(settings, 'USE_PROMOS', True)
+    use_promo = getattr(settings, 'USE_PROMOS', True)
+    show_promo = project.allow_promos
     gold_user = gold_project = False
+    promo_obj = None
+
     # User is a gold user, no promos for them!
     if request.user.is_authenticated():
         if request.user.gold.count() or request.user.goldonce.count():
-            show_promo = False
             gold_user = True
-    # Explicit promo disabling
-    if project.slug in getattr(settings, 'DISABLE_PROMO_PROJECTS', []):
-        show_promo = False
     # A GoldUser has mapped this project
-
     if project.gold_owners.count():
-        show_promo = False
         gold_project = True
 
-    promo_obj = SupporterPromo.objects.filter(live=True, display_type='doc').order_by('?').first()
-    if not promo_obj:
+    if gold_user or gold_project:
         show_promo = False
 
-    # Support showing a "Thank you" message for gold folks
-    if gold_user:
-        gold_promo = SupporterPromo.objects.filter(live=True, name='gold-user')
-        if gold_promo.count():
-            promo_obj = gold_promo.first()
-            show_promo = True
-
-    # Default to showing project-level thanks if it exists
-    if gold_project:
-        gold_promo = SupporterPromo.objects.filter(live=True, name='gold-project')
-        if gold_promo.count():
-            promo_obj = gold_promo.first()
-            show_promo = True
+    if use_promo and show_promo:
+        promo_obj = (SupporterPromo.objects
+                     .filter(live=True, display_type='doc')
+                     .order_by('?')
+                     .first())
+
+        # Support showing a "Thank you" message for gold folks
+        if gold_user and promo_obj is None:
```

Comment:
Oops, yeah that's a logic error, I was considering the reverse case -- where if there is no ad and user is gold, show a gold promo. This doesn't make sense of course. :)